components/vr-skybox.js

```diff
@@ -0,0 +1,57 @@
+/* globals define */
+(function(define){'use strict';define(function(require,exports,module){
+
+  document.registerElement(
+    'vr-skybox',
+    {
+      prototype: Object.create(
+        VRObject.prototype, {
+          init: {
+            value: function() {
+
+              var self = this;
+              var size = parseFloat(this.getAttribute('size')) || 10000;
+              
+              var urlPrefix = this.getAttribute('src');
+              var urls = [ urlPrefix + "right.jpg", urlPrefix + "left.jpg",
+                  urlPrefix + "top.jpg", urlPrefix + "bottom.jpg",
+                  urlPrefix + "front.jpg", urlPrefix + "back.jpg" ];
+              
+              var textureCube = THREE.ImageUtils.loadTextureCube( urls, THREE.CubeReflectionMapping, function(){ 
+                self.load(); 
+              });
+              textureCube.format = THREE.RGBFormat;
+              var shader = THREE.ShaderLib['cube'];
+              var material = new THREE.ShaderMaterial({
+                fragmentShader: shader.fragmentShader,
+                vertexShader: shader.vertexShader,
+                uniforms: shader.uniforms,
+                side: THREE.BackSide
+              });
+
+              material.uniforms[ "tCube" ].value = textureCube;
+              var geometry = new THREE.BoxGeometry( size, size, size, 1, 1, 1 )
+
+              if (!this.object3D) {
```

Comment:
This is the init function. No need to check for this.object3D. You can simply do 
```javascript
 this.object3D = new THREE.Mesh( geometry, material );
```